File: frontend/src/types/lesson.ts
/** 五步闯关步骤枚举 */
export type LessonStep =
    | 'IMPORT'
    | 'EXPLAIN'
    | 'EXAMPLE'
    | 'PRACTICE'
    | 'SUMMARY'
    | 'COMPLETED';

/** 闯关步骤显示信息 */
export const LESSON_STEP_META: Record<LessonStep, { label: string; icon: string; order: number }> = {
    IMPORT: { label: '基础预热', icon: '🔥', order: 0 },
    EXPLAIN: { label: '深入讲解', icon: '📖', order: 1 },
    EXAMPLE: { label: '典型例题', icon: '✏️', order: 2 },
    PRACTICE: { label: '上手实操', icon: '🎯', order: 3 },
    SUMMARY: { label: '总结复盘', icon: '🏆', order: 4 },
    COMPLETED: { label: '已完成', icon: '✅', order: 5 },
};

/** 闯关状态响应 — 对齐后端 LessonStatusResponse */
export interface LessonStatusResponse {
    lesson_id: string;
    student_id?: string;
    node_id?: string;
    material_id?: string;
    current_step: LessonStep;
    is_completed: boolean;
    node_title?: string;
    content_preview?: string;
    step_prompt?: string;
    message?: string;
    error?: string;
}

/** 学习计划任务类型 */
export type TaskType = 'LEARN_NEW' | 'DO_QUIZ' | 'REVIEW_VARIANT';

export interface PlanItem {
    id: string;
    type: TaskType;
    title: string;
    completed: boolean;
    duration_min: number;
    date: string; // YYYY-MM-DD
}

export interface PlanListResponse {
    student_id: string;
    items: PlanItem[];
}

File: backend/app/schemas/lesson.py
"""
Lesson Schemas - Pydantic models for guided learning API
"""

from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class LessonStartRequest(BaseModel):
    """Request to start or resume a guided learning session."""

    student_id: str = Field(..., description="Student's unique ID")
    node_id: str = Field(..., description="Knowledge node ID to study")


class LessonAdvanceRequest(BaseModel):
    """Request to advance to the next lesson step."""

    student_id: str
    node_id: str


class LessonStatusResponse(BaseModel):
    """Response with current lesson status."""

    lesson_id: str
    student_id: Optional[str] = None
    node_id: Optional[str] = None
    material_id: Optional[str] = None
    current_step: str
    is_completed: bool = False
    node_title: Optional[str] = None
    content_preview: Optional[str] = None
    step_prompt: Optional[str] = None
    message: Optional[str] = None
    error: Optional[str] = None


class PlanGenerateRequest(BaseModel):
    """Request to generate studying plans via Planner Agent."""

    student_id: str
    material_id: str
    start_date: Optional[str] = None
    sessions_per_week: Optional[int] = 3


class PlanItemResponse(BaseModel):
    """A single scheduled study task."""

    id: str
    node_id: str
    type: str  # 'LEARN_NEW' or 'DO_QUIZ' or 'REVIEW_VARIANT'
    title: str
    completed: bool
    duration_min: int
    date: str


class PlanListResponse(BaseModel):
    """List of planned study tasks for the student."""

    student_id: str
    items: list[PlanItemResponse]
    start_date: str | None = None
    end_date: str | None = None
